wp-includes/parser/class-wp-parser-node.php

```diff
@@ -102,83 +102,169 @@ public function merge_fragment( $node ) {
 		$this->children = array_merge( $this->children, $node->children );
 	}
 
-	public function has_child( $rule_name ) {
+	public function has_child(): bool {
+		return count( $this->children ) > 0;
+	}
+
+	public function has_child_node( ?string $rule_name = null ): bool {
 		foreach ( $this->children as $child ) {
-			if ( ( $child instanceof WP_Parser_Node && $child->rule_name === $rule_name ) ) {
+			if (
+				$child instanceof WP_Parser_Node
+				&& ( null === $rule_name || $child->rule_name === $rule_name )
+			) {
 				return true;
 			}
 		}
 		return false;
 	}
 
-	public function has_token( $token_id = null ) {
+	public function has_child_token( ?int $token_id = null ): bool {
 		foreach ( $this->children as $child ) {
-			if ( $child instanceof WP_MySQL_Token && (
-				null === $token_id ||
-				$child->type === $token_id
-			) ) {
+			if (
+				$child instanceof WP_Parser_Token
+				&& ( null === $token_id || $child->id === $token_id )
+			) {
 				return true;
 			}
 		}
 		return false;
 	}
 
-	public function get_token( $token_id = null ) {
+
+	public function get_child() {
```

Comment:
Let's call it `get_first_child()` to clarify what happens when there's more than one